test/base.py

```diff
@@ -52,7 +52,9 @@ def config_template(self, override=None):
         # NOTE: we set pg_receivexlog_path and pg_basebackup_path per site and globally mostly to verify that
         # it works, the config keys are deprecated and will be removed in a future release at which point we'll
         # switch to using pg_bin_directory config.
-        bindir, ver = find_pg_binary("")
+        pgexe, ver = find_pg_binary("postgres")
+        if pgexe is not None:
+            bindir = os.path.dirname(pgexe)
```

Comment:
If there is no bindir defined, it will error out later in ` "pg_receivexlog_path": os.path.join(bindir, "pg_receivexlog"),` -> another reason to convert the function to "success or exception" :-)

Reply:
Done.